### ISSUE TITLE
ubootwrite

### ISSUE THREAD

**Issue by HorstBaerbel** [rvalles/brntool#1]
I found no other way to reach you. This is just to let you know [ubootwrite](https://github.com/HorstBaerbel/ubootwrite) exists. I hope you're ok with the "fork"... If not, please let me know.

Best Regards,

Bim

**Comment by rvalles** [comment]
Hi Bim,

I'm actually quite happy to see the project is somehow alive, and I'm definitely OK with forks (hence the GPLv3 choice).

I'm however not amused regarding attribution. Readme mentions the prior fork (which I didn't even know existed, I'm happy to read about that too) author, but I don't get the same treatment, which doesn't seem right.

And (sort of suspected), there's blank lines padding everywhere (ouch!). This one is a losing battle I guess, I'll live with it :)

Thanks for letting me know

Kind Regards,
Roc Vallès Domènech

**Comment by HorstBaerbel** [comment]
Hmmm. Sorry. I linked to brntool twice, just to make sure...
About the blank lines: It's basically copy-paste from a forum and I'm a total Python noob btw... :) I have to admit though I was being lazy. I could have properly forked brntool and done the work "pgid69" saved himself... ;)
How about you adopting the script/code? I saw the "--write" option is commented in brntool. Could you intergrate the functionality into brntool (the transfer rate / eta stuff is definitely useful) or just add it standalone as "brnwrite.py"? If you don't want to, I can do the fork, adjust brntool and send you a pull request.
But I got absolutely no problems with simply deleting my repo and acting as a beta tester if need be...

Btw. I tested brntool (for dumping the firmware) and ubootwrite on an Arcadyan ARV752DPW22 (Easybox 803A) router and it worked like a charm a couple of times. Thanks!

Kind Regards,

Bim

**Comment by pgid69** [comment]
Hello

I'm pgid69, 

I've just found this issue where my name is mentionned.
You say that you regret that i don't contact you when i use the code you write to add a new functionnality.
You're right and i apologize for this.

But I just want to say that it was not my intention to make an official fork.
At first, as I explained in this [post](https://forum.openwrt.org/viewtopic.php?pid=183315#p183315)  i just wanted to load a new bootloader in my modem.
I wrote a quick and dirty adaptation of your code because it handles the serial port, and when i saw that the code worked, i decided to publish on the forum, because i thought that it could help other people. Even if i knew that the code i wrote was not perfect as it was my first python script.

I never had any intention to hide your contribution to the code.
And i want to thank you, even if it's late, for the code you wrote.

Best regards

Gilles
